pinot-core/src/main/java/org/apache/pinot/core/segment/name/SegmentNameGenerator.java

```diff
@@ -18,13 +18,23 @@
  */
 package org.apache.pinot.core.segment.name;
 
-import org.apache.pinot.core.segment.creator.ColumnStatistics;
+import com.google.common.base.Joiner;
+import javax.annotation.Nullable;
 
 
 /**
- * An interface that allows generates names for segments depending on the naming scheme.
+ * Interface for segment name generator based on the segment sequence id and time range.
  */
 public interface SegmentNameGenerator {
-  String generateSegmentName(ColumnStatistics timeColStatsCollector)
-      throws Exception;
+  Joiner JOINER = Joiner.on('_').skipNulls();
+
+  /**
+   * Generates the segment name.
+   *
+   * @param sequenceId Segment sequence id (negative value means INVALID)
+   * @param minTimeValue Minimum time value
+   * @param maxTimeValue Maximum time value
+   * @return Segment name generated
+   */
+  String generateSegmentName(int sequenceId, @Nullable Object minTimeValue, @Nullable Object maxTimeValue);
```

Comment:
+1 on moving `sequenceId` to `generateSegmentName` instead of passing it via constructor...